### ISSUE TITLE
RV1805_CAP_RC is not used but should be

### ISSUE THREAD

**Issue by kpfleming** [sparkfun/SparkFun_RV-1805_Arduino_Library#11]
The BOB-14558 Qwiic board for this device includes the recommended (by the application manual) 47pf capacitor connected between the Cap_RC pin and Vss. However, this library does not write the necessary value to RV1805_CAP_RC to enable the capacitor to be used, so the autocalibrated RC oscillator will not be as accurate as it could be. Since the hardware is present, the guidance in 4.11.5 of the [app manual](https://www.microcrystal.com/fileadmin/Media/Products/RTC/App.Manual/RV-1805-C3_App-Manual.pdf) should be followed to enable its usage.

**Comment by adamgarbo** [comment]
I'm currently conducting accuracy tests of the RV-1805 under varying temperatures and I'm curious if this could be why the RTC is drifting so much (~10 seconds over 12 hours). 

Enabling the CAP_RC can be accomplished manually with two register writes:

```
// Write 9Dh to CONFKEY Register (1Fh) to enable write access to the CAPRC Register (26h)
rtc.writeRegister(0x1F, 0x9D);

// Write A0h to CAPRC register to enable Cap_RC pin
rtc.writeRegister(0x26, 0xA0); 
```

For the library, the code would be as follows:

```
writeRegister(RV1805_CONF_KEY, RV1805_CONF_WRT); // Enable write access to the CAPRC Register (26h)
writeRegister(RV1805_CAP_RC, 0xA0); // Enable Cap_RC pin
```

I can submit a PR if needed.

Cheers,
Adam

**Comment by kpfleming** [comment]
Thanks for that note: I don't have a practical way to test the device's accuracy, so I'm glad someone else out there does.

**Comment by adamgarbo** [comment]
Hi there,

I did a brief test of the RV-1805 today with the Cap_RC pin enabled and saw no improvement in the drift of the RTC over time.

[https://i.imgur.com/1StVxi8.png](https://i.imgur.com/1StVxi8.png)

In a previous 24-hour test I did of the SparkFun Artemis Nano and Qwiic RV-1805, I observed the same drift over time in the RV-1805. In both tests, the RTCs were compared against the Maxim DS3231SN, which has an accuracy of ±3.0 ppm over the range of -40°C to +85°C. The RV-1805 **_should_** be performing better than this, so I wonder what the issue could be.

[https://i.imgur.com/VLqNMoq.png](https://i.imgur.com/VLqNMoq.png)

**Comment by AndyEngland521** [comment]
@adamgarbo have you checked register 0x14? A while back we received a batch of RTC's from the manufacturer that didn't have this calibration register set.

**Comment by adamgarbo** [comment]
Hey @AndyEngland521 , 

That was one of the first places I checked. The batch of RV-1805s I have (10+) all have Bit 7 register 0x14 set to 0 (XT calibration adjust mode: Normal Mode). I anticipated that there would be accuracy issues below -10°C as this is beyond the autocalibration temperature range, but I'll keep trying to figure out where this drift is coming from.

It would be great to see SparkFun offer a low-power, temperature-compensated RTC such as the RV-3129-C3 ([https://www.microcrystal.com/en/products/real-time-clock-rtc/rv-3129-c3/](https://www.microcrystal.com/en/products/real-time-clock-rtc/rv-3129-c3/)). Do you think this could be a possibility? 

Cheers,
Adam

**Comment by AndyEngland521** [comment]
@adamgarbo Is the rest of the 0x14 register zeroed out as well? This was where we had an issue with the manufacturer as they were supposed to calibrate this register in the factory.

I agree with your sentiment for a low-power, temperature-compensated RTC, we're considering something similar.

**Comment by adamgarbo** [comment]
@AndyEngland521 the entire 0x14 register reads as: 0b01101110. Looks like the factory calibration is present.

Very keen to hear more about what you may be considering!

**Comment by AndyEngland521** [comment]
Well that's a good sign, have you checked the Auto-Calibration Failure bit (bit 0 in register 0x1D) which will go to 1 if the RTC is unable to perform auto-calibration? if it is set to 1, clear it and watch to see if it gets set again.

**Comment by adamgarbo** [comment]
@AndyEngland521  I'm currently testing the RV-1805 at cold temperatures while monitoring the serial output. 

I noticed that OMODE (bit 5) in the 0x1D register was set to 1 (The RC Oscillator is selected to drive the internal clocks.), and that OSEL (bit 7) in the 0x1C register was also set to 1 (Request the RC Oscillator to generate the clock for the timer circuits ). I've since set OSEL in 0x1C to 0 (Request the XT Oscillator to generate a 32.768 kHz clock to the timer circuit) and confirmed in 0x1D that OMODE is now 0 (The XT Oscillator is selected to drive the internal clocks.). I'm basically forcing it to operate in XT Autocalibration Mode and will keep an eye on bits 0 & 1 in 0x1D to see if there are any Autocalibration or XT oscillator failures. I'll also test to see if the accuracy improves.

I wonder if the RV-1805 was stuck in RC Autocalibration mode, but at the same time dealing with the Cap_RC pin not being set high. Time will tell! (har har)

**Comment by adamgarbo** [comment]
@AndyEngland521 

So, it looks like the RV-1805 board I was previously testing with was faulty. In the test described above, I noticed that the XT Oscillator Failure Interrupt was constantly being triggered (bit 1 in the 0x1C register was being set to 1). Each time I'd clear the bit, it would trigger again. So, I swapped in a new board, ran another test overnight and had **_much_** better results: https://i.imgur.com/lxEPgVz.png It still drifts at cold temperatures, but that's to be expected.

It's clear that the earlier accuracy errors were due to a bad board, but the Cap_RC pin should still be enabled in the library with the two lines below. Would begin() be the most appropriate place for them?

Cheers,
Adam

```
writeRegister(RV1805_CONF_KEY, RV1805_CONF_WRT); // Enable write access to the CAPRC Register (26h)
writeRegister(RV1805_CAP_RC, 0xA0); // Enable Cap_RC pin
```

**Comment by AndyEngland521** [comment]
@adamgarbo I'd say yes to that! I can create a new release later in the morning as currently I'm on mobile setting up my home work station.

**Comment by adamgarbo** [comment]
Created a PR for you. No rush and thanks for the help! [https://github.com/sparkfun/SparkFun_RV-1805_Arduino_Library/pull/14](url)

**Comment by AndyEngland521** [comment]
Fixed in v1.1.0

Thanks to @adamgarbo for the PR
